src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1128,5 +1128,24 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
             }
             Output(SpanKind.Markup);
         }
+
+        internal static bool IsValidAttributeNameSymbol(HtmlSymbol symbol)
+        {
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            var symbolType = symbol.Type;
```

Comment:
Long list and perhaps the `... == ... || ...` list would be (a bit) longer. Nonetheless suggest switching to the positive list for those of us who haven't memorized `HtmlSymbolType`.